clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -343,40 +408,123 @@ export class CopybookDownloadService {
     if (this.handleAsEndevorElement(documentUri)) {
       return !!(await this.e4eDownloader?.getE4EConfig(documentUri));
     }
-    if (!this.explorerApi) return false;
 
-    const copybooksLocation =
-      DownloadUtil.areCopybookDownloadConfigurationsPresent(
-        documentUri,
-        dialects,
-      );
+    const configs = await loadProcessorGroupCopybookPathsConfig(
+      { scopeUri: documentUri },
+      [],
+    );
 
-    if (!copybooksLocation) {
-      return false;
-    }
+    const procGroupZoweConfigs = configs.filter(
+      (config): config is ZoweUssConfigModel | ZoweDatasetConfigModel =>
+        typeof config != "string" && (DATASET in config || USS in config),
+    );
+
+    const endevorConfigs = configs.filter(
+      (config): config is EndevorConfigModel =>
+        typeof config != "string" && ENVIRONMENT in config,
+    );
 
     const profile = ProfileUtils.getProfileNameForCopybook(
       documentUri,
       this.explorerApi,
     );
-    const availableProfiles = ProfileUtils.getAvailableProfiles(
-      this.explorerApi,
-    );
-    if (!profile || !availableProfiles.includes(profile)) {
-      const message = profile
-        ? `${PROVIDE_PROFILE_MSG} Provided invalid profile name: ${profile}`
-        : `${PROVIDE_PROFILE_MSG}`;
-      this.processDownloadError(message);
+
+    if (endevorConfigs.length > 0 && !this.e4eApi) {
+      this.diagnosticsService?.showDiagnostics(vscode.Uri.parse(documentUri), [
+        {
+          range: new vscode.Range(
+            new vscode.Position(0, 0),
+            new vscode.Position(1, 0),
+          ),
+          message: "Explorer for Endevor is not installed",
+          severity: vscode.DiagnosticSeverity.Warning,
+        },
+      ]);
```

Comment:
```suggestion
      this.missingExtension(documentUri, "Explorer for Endevor is not installed");
```
and similarly below.